Water sensors code/IoT Device/example.c

```diff
@@ -0,0 +1,224 @@
+ * limitations under the License.
+ */
+
+#include <common.h>
+#include <string.h>
+#include <math.h>
+
+#include <platform.h>
+#include <gsm.h>
+#include <pin_mux.h>
+#include <leds.h>
+#include <powerControl.h>
+#include <client_api.h>
+#include <base64_codec_transport.h>
+#include <thingstream_transport.h>
+#include <modem_transport.h>
+#include <client_platform.h>
+#include <log_client_transport.h>
+#include <log_modem_transport.h>
+#include <line_buffer_transport.h>
+#include <gsm_uart_transport.h>
+#include <debug_printf.h>
+
+static uint8_t gsm_buffer[290];
+
+/* This routine is called after an ASSERT() macro has failed.
+ * @param line  the line number of the ASSERT() failure.
+ * @param what  the text of the ASSERT() being checked.
+ * @return loops indefinitely so does not return.
+ */
+static void AssertFail(int line, const char* what)
+{
+    debug_printf("Assert fail \"%s\" at line %d\n", what, line);
+    Led_Flash(100, 0, 0, 500, 500, 10);
+    Led_Background(100, 0, 0);
+    while (true)
+    {
+        CheckForSysTick();
+        __WFI(); /* Wait for Interrupt */
+    }
+}
+#define ASSERT(truth)                                                        \
+    do                                                                       \
+    {                                                                        \
+        int line = __LINE__;                                                 \
+        if (!(truth))                                                        \
+        {                                                                    \
+            AssertFail(line, #truth);                                        \
+        }                                                                    \
+    } while(0)
+
+static void dump_payload(uint8_t* payload, uint16_t payloadlen)
+{
+    DEBUGOUT("\"");
+    while (payloadlen-- > 0)
+    {
+        uint8_t byte = *payload++;
+        if (byte == '\"')
+        {
+            DEBUGOUT("\\\"");
+        }
+        else if ((byte >= ' ') && (byte <= '~'))
+        {
+            DEBUGOUT("%c", byte);
+        }
+        else if (byte == '\n')
+        {
+            DEBUGOUT("\n");
+        }
+        else
+        {
+            DEBUGOUT("[%02x]", byte);
+        }
+    }
+    DEBUGOUT("\"\n");
+}
+
+static void subscribe_callback(void *cookie, Topic topic, QOS qos, uint8_t* payload, uint16_t payloadlen)
+{
+    DEBUGOUT("subscribe_callback: topic %d::%d ",
+             topic.topicType, topic.topicId);
+    dump_payload(payload, payloadlen);
+}
+
+#define THINGSTREAM_BUFFER_LENGTH 1024
+static uint8_t thingstream_buffer[THINGSTREAM_BUFFER_LENGTH];
+
+
+static void modem_callback(void *cookie, const char* response, uint16_t len)
+{
+    DEBUGOUT("modem_callback: ");
+    dump_payload((uint8_t*)response, len);
+}
+
+int getFinalValue(int vals[]){
+	double value = pow(2,4)*vals[0]+pow(2,3)*vals[1]+pow(2,2)*vals[2]+2*vals[3]+1*vals[4];
```

Comment:
This calculation needs parenthesis to ensure the right order of execution

Reply:
why, Arithmetic ?

Reply:
Well first of it will increase readability if it is correctly structured. 
It will make it easier to maintain when changes are required 
If the execution order is wrong the calculation will be wrong

Reply:
alright, I hear you. Made the changes as per your request